akka-remote/src/test/scala/akka/remote/artery/LargeMessagesChannelSpec.scala

```diff
@@ -0,0 +1,157 @@
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(RegularDestination)
+
+      } finally {
+        systemA.terminate()
+        systemB.terminate()
+      }
+    }
+
+    "pass small regular messages over the large-message channel" in {
+      val Seq(portA, portB) = SocketUtil.temporaryServerAddresses(2, "localhost", udp = true).map(_.getPort)
+      val systemA = ActorSystem("systemA", config(portA))
+      val systemB = ActorSystem("systemB", config(portB))
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // start actor and make sure it is up and running
+        val large = systemB.actorOf(Props(new EchoSize), "large")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // communicate with it from the other system
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "large"))
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(LargeDestination)
+
+      } finally {
+        systemA.terminate()
+        systemB.terminate()
+      }
+    }
+
+    "allow for normal communication while simultaneously sending large messages" in {
+      val Seq(portA, portB) = SocketUtil.temporaryServerAddresses(2, "localhost", udp = true).map(_.getPort)
+      val systemA = ActorSystem("systemA", config(portA))
+      val systemB = ActorSystem("systemB", config(portB))
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // setup two actors, one with the large flag and one regular
+        val large = systemB.actorOf(Props(new EchoSize), "large")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        val regular = systemB.actorOf(Props(new EchoSize), "regular")
+        regular.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // both up and running, resolve remote refs
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "large"))
+        val regularRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "regular"))
+
+        // send a large message, as well as regular one
+        val largeBytes = 500000
+        largeRemote.tell(Ping(ByteString.fromArray(Array.ofDim[Byte](largeBytes))), senderProbeA.ref)
+        regularRemote.tell(Ping(), senderProbeA.ref)
```

Comment:
I don't get this, doesn't a selection always boil down to an `ActorRef` before `Association.send` is invoked anyways?